evm/src/cpu/kernel/asm/bignum/mul.asm

```diff
@@ -0,0 +1,53 @@
+// Arithmetic on little-endian integers represented with 128-bit limbs.
+// All integers must be under a given length bound, and are padded with leading zeroes.
+
+// Stores a * b in output_loc, leaving a and b unchanged.
+// Both a and b have length len; a * b will have length 2 * len.
+// output_loc must be initialized as 2 * len zeroes.
```

Comment:
Hmm interesting — but that will make the code more bloated, both in terms of complexity and in terms of kernel size. That initialization will still be complicated because there still needs to be the add-and-carry dynamic from `addmul`. So I'm not sure it's worth it?

Reply:
Yeah, it's a trade-off; the change will indeed increase the kernel size, but it will reduce the number of instructions executed per kernel invocation. But at this stage I agree, probably not worth it, especially if it's not a burden for the calling code to guarantee that the memory is initialised to zero before every call.

Reply:
I'll leave a TODO for now!